tracker-s2bot/ftnoir_tracker_s2bot.cpp

```diff
@@ -0,0 +1,114 @@
+#include "ftnoir_tracker_s2bot.h"
+#include "api/plugin-api.hpp"
+
+#include <cinttypes>
+#include <algorithm>
+#include <cmath>
+#include <QNetworkRequest>
+#include <QNetworkReply>
+
+tracker_s2bot::tracker_s2bot() : pose { 0,0,0, 0,0,0 }, m_nam (std::make_unique<QNetworkAccessManager>())
+{
+}
+
+tracker_s2bot::~tracker_s2bot()
+{
+    requestInterruption();
+    wait();
+}
+
+template<typename t>
+static const t bound(t datum, t least, t max)
+{
+    if (datum < least)
+        return least;
+    if (datum > max)
+        return max;
+    return datum;
+}
+
+void tracker_s2bot::run() {
+#pragma pack(push, 1)
+    struct {
+        uint8_t pad1;
+        uint8_t flags;
+        float fl[12];
+    } data;
+#pragma pack(pop)
+    enum F {
+        flag_Raw = 1 << 0,
+        flag_Orient = 1 << 1,
+        Mask = flag_Raw | flag_Orient
+    };
+
+	if (s.freq == 0) s.freq = 10;
+	timer.setInterval(1000.0/s.freq);
+	timer.setSingleShot(false);	
+	connect(&timer, &QTimer::timeout, [this]() {
+		auto reply = m_nam->get(QNetworkRequest(QUrl("http://localhost:17317/poll")));
```

Comment:
A http request adds significant overhead. Please note that camera tracking does 187 Hz with no worry, and IMU tracker does 250 Hz, same as all Oculus Rift versions. Have you tried it in some flight/other simulation software? How bad is it?